server/sockets.js

```diff
@@ -567,31 +590,14 @@ if (cluster.isMaster) {
 			room = rooms.get(roomid);
 			if (!room) return;
 
-			/** @type {[string?, string?, string?]} */
-			let messages = [null, null, null];
+			/** @type {[string?, string?, string?, string?, string?]} */
+			const messages = [null, null, null, null, null];
 			message = data.substr(nlLoc + 1);
 			roomChannel = roomChannels.get(roomid);
 			for (const [socketid, socket] of room) {
-				switch (roomChannel ? roomChannel.get(socketid) : '0') {
-				case '1':
-					if (!messages[1]) {
-						messages[1] = message.replace(/\n\|split\n[^\n]*\n([^\n]*)\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-					}
-					socket.write(messages[1]);
-					break;
-				case '2':
-					if (!messages[2]) {
-						messages[2] = message.replace(/\n\|split\n[^\n]*\n[^\n]*\n([^\n]*)\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-					}
-					socket.write(messages[2]);
-					break;
-				default:
-					if (!messages[0]) {
-						messages[0] = message.replace(/\n\|split\n([^\n]*)\n[^\n]*\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-					}
-					socket.write(messages[0]);
-					break;
-				}
+				const channelid = roomChannel ? Number(roomChannel.get(socketid)) : 0;
+				if (!messages[channelid]) messages[channelid] = extractChannel(message, channelid);
```

Comment:
May need to be `messages[channelid + 1]` under the new definition of `channelid`.

Reply:
Hmm, The old code was `'1'`, `'2'`, and `default`, with no `roomChannel` becoming `0`, where `0` was actually grabbing the first of the 4 splits (which corresponds to spectator, as channelid = 0 still does). What makes you think it should be +1?

Reply:
`channelid` is currently an int ranging from `-1` to `4`, right? `+1` makes that `0` to `5`, allowing it to be an array index. `messages` is a length-6 array, after all.

Reply:
Are we reading the same code? `messages` is 5 elements (was previously 3 under the 4 stream system). I also don't think `channelid` is an int from -1 to 4? `roomChannel.get(socketid)` seems to return a string `'1'` or `'2'` under the old code?

Reply:
You're right, `channelid` is a string, which makes it even less suited for being stuffed into a key of `messages` – I'm surprised TypeScript isn't complaining.

Reply:
Its not a string anymore - I `Number` it on purpose now.

Reply:
Oh wait no it's being cast to `Number`, no wonder it works here.